nilearn/input_data/base_masker.py

```diff
@@ -38,6 +40,40 @@ def filter_and_mask(imgs, mask_img_,
 
     mask_img_ = _utils.check_niimg(mask_img_, ensure_3d=True)
     imgs = _utils.check_niimgs(imgs, accept_3d=True)
+    if mask_time is None:
+        imgs = _utils.check_niimgs(imgs, accept_3d=True)
+    else:
+        # check type and value of mask_time
+        if (not isinstance(mask_time, list) or
+            len(mask_time) == 0 or
+            not isinstance(mask_time[0], bool)):
```

Comment:
> Indeed, it is unclear for me what are the use cases addressed here that
> are not also addressed in the indexing/slicing PR ?

The indexing/slicing PR does make things much easier. However, suppose
that you want, eg to remove the 3 first volumes. With Danilo's PR, it is
trivial to do this in a pipeline, in distributed computing. Manually
using slicing would require more work.

However, the slicing PR should be merged, and then this PR updated to use
it.